src/Uno.UI/ts/WindowManager.ts

```diff
@@ -1598,6 +1583,136 @@ namespace Uno.UI {
+			const  e = this.getView(elementId);
+			if (e instanceof SVGElement) {
+
+				e.style.fill = this.getCssColorOrUrlRef(fillColor, fillPaintRef);
+				e.style.stroke = this.getCssColorOrUrlRef(strokeColor, strokePaintRef);
+			}
+		}
+
+		public setShapeStylesFast2(elementId: number, fillColor: number, fillPaintRef: number, strokeColor: number, strokePaintRef: number, strokeWidth: number, strokeDashArray: any[]): void {
+			const  e = this.getView(elementId);
+			if (e instanceof SVGElement) {
+
+				e.style.fill = this.getCssColorOrUrlRef(fillColor, fillPaintRef);
+				e.style.stroke = this.getCssColorOrUrlRef(strokeColor, strokePaintRef);
+				e.style.strokeWidth = `${strokeWidth}px`;
+				e.style.strokeDasharray = strokeDashArray.join(',');
+			}
+		}
+
+		public setSvgFillRule(htmlId: number, nonzero: boolean): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGPathElement) {
+				e.setAttribute('fill-rule', nonzero ? 'nonzero' : 'evenodd');
+			}
+		}
+
+		public setSvgEllipseAttributes(htmlId: number, cx: number, cy: number, rx: number, ry: number): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGEllipseElement) {
+				e.setAttribute('cx', cx.toString());
+				e.setAttribute('cy', cy.toString());
+				e.setAttribute('rx', rx.toString());
+				e.setAttribute('ry', ry.toString());
+			}
+		}
+
+		public setSvgLineAttributes(htmlId: number, x1: number, x2: number, y1: number, y2: number): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGLineElement) {
+				e.setAttribute('x1', x1.toString());
+				e.setAttribute('x2', x2.toString());
+				e.setAttribute('y1', y1.toString());
+				e.setAttribute('y2', y2.toString());
+			}
+		}
+
+		public setSvgPathAttributes(htmlId: number, nonzero: boolean, data: string): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGPathElement) {
+				e.setAttribute('fill-rule', nonzero ? 'nonzero' : 'evenodd');
+				e.setAttribute('d', data);
+			}
+		}
+
+		public setSvgPolyPoints(htmlId: number, points: number[]): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGPolygonElement || e instanceof SVGPolylineElement) {
+				if (points != null) {
+					const delimiters = [' ', ','];
+					// interwave to produce: x0,y0 x1,y1 ...
+					// i start at 1
+					e.setAttribute('points', points.reduce((acc, x, i) => acc + delimiters[i % delimiters.length] + x, ''));
+				}
+				else {
+					e.removeAttribute('points');
+				}
+			}
+		}
+
+		public setSvgRectangleAttributes(htmlId: number, x: number, y: number, width: number, height: number, rx: number, ry: number): void {
+			const e = this.getView(htmlId);
+			if (e instanceof SVGRectElement) {
+				e.x.baseVal.value = x;
+				e.y.baseVal.value = y;
+				e.width.baseVal.value = width;
+				e.height.baseVal.value = height;
+
+				e.setAttribute('rx', rx.toString());
```

Comment:
in the older code, we were using setAttribute (via setAttributesNativeFast)
but it seems that we could use .baseVal.value here also
but whats the ramification of that, and which to favor?

Reply:
If we can avoid using strings, why not, I don't think of any other ramifications.